core/templates/dev/head/domain/editor/undo_redo/UndoRedoService.js

```diff
@@ -113,6 +113,18 @@ oppia.factory('UndoRedoService', [
       return committableChangeList;
     };
 
+    UndoRedoService.setBackendChangeList = function(changeList) {
```

Comment:
I'm mostly confused about this. It's not "setting" -- it's deleting some stuff from the incoming changelist and then setting.

This means there are side effects to the incoming input, which is not great.

It would be good to find a better/clearer way to do the technical design here.

Reply:
Done, not modifying the incoming changelist.